packages/core/addon/models/bard-request-v2/fragments/base.js

```diff
@@ -21,4 +23,72 @@ export default class Base extends Fragment.extend(Validations) {
     }
   })
   parameters;
+
+  @service('bard-metadata')
+  metadataService;
+
+  _meta;
+
+  /**
+   * @type {Meta}
+   */
+  get meta() {
+    if (!this._meta) {
+      if (this.lookupField === 'dateTime') {
+        this._meta = { id: 'dateTime', name: 'Date Time' };
+        return this._meta;
```

Comment:
Can we avoid side-effects in the getter? Are we trying to cache? If so, throw a `@computed` on it.

Reply:
Or move the compute to another property

Reply:
Another thought, instead of doing lookup here when loading a request, and using the assigned metadata via factory creation, could we do this lookup in the fragment serializer? That way, when we get to this object, there is just one way to get the metadata.

Reply:
Yeah it's not just cache, it's more to do with not having to do parent model lookup if the meta is already there.

I thought attaching at serialization/transform time was exactly what we were trying not to do, and trying to keep this lazy.